Userland/Libraries/LibWeb/Cookie/ParsedCookie.cpp

```diff
@@ -151,92 +183,139 @@ void process_attribute(ParsedCookie& parsed_cookie, StringView attribute_name, S
     }
 }
 
+static constexpr AK::Duration maximum_cookie_age()
+{
+    return AK::Duration::from_seconds(400LL * 24 * 60 * 60);
+}
+
+// https://www.ietf.org/archive/id/draft-ietf-httpbis-rfc6265bis-15.html#section-5.6.1
 void on_expires_attribute(ParsedCookie& parsed_cookie, StringView attribute_value)
 {
-    // https://tools.ietf.org/html/rfc6265#section-5.2.1
-    if (auto expiry_time = parse_date_time(attribute_value); expiry_time.has_value())
-        parsed_cookie.expiry_time_from_expires_attribute = expiry_time.release_value();
+    // 1. Let the expiry-time be the result of parsing the attribute-value as cookie-date (see Section 5.1.1).
+    auto expiry_time = parse_date_time(attribute_value);
+
+    // 2. If the attribute-value failed to parse as a cookie date, ignore the cookie-av.
+    if (!expiry_time.has_value())
+        return;
+
+    // 3. Let cookie-age-limit be the maximum age of the cookie (which SHOULD be 400 days in the future or sooner, see
+    //    Section 5.5).
+    auto cookie_age_limit = UnixDateTime::now() + maximum_cookie_age();
+
+    // 4. If the expiry-time is more than cookie-age-limit, the user agent MUST set the expiry time to cookie-age-limit
+    //    in seconds.
+    if (expiry_time->seconds_since_epoch() > cookie_age_limit.seconds_since_epoch())
+        expiry_time = cookie_age_limit;
+
+    // 5. If the expiry-time is earlier than the earliest date the user agent can represent, the user agent MAY replace
+    //    the expiry-time with the earliest representable date.
+    if (auto earliest = UnixDateTime::earliest(); *expiry_time < earliest)
+        expiry_time = earliest;
+
+    // 6. Append an attribute to the cookie-attribute-list with an attribute-name of Expires and an attribute-value of
+    //    expiry-time.
+    parsed_cookie.expiry_time_from_expires_attribute = expiry_time.release_value();
 }
 
+// https://www.ietf.org/archive/id/draft-ietf-httpbis-rfc6265bis-15.html#section-5.6.2
 void on_max_age_attribute(ParsedCookie& parsed_cookie, StringView attribute_value)
 {
-    // https://tools.ietf.org/html/rfc6265#section-5.2.2
+    // 1. If the attribute-value is empty, ignore the cookie-av.
+    if (attribute_value.is_empty())
+        return;
 
-    // If the first character of the attribute-value is not a DIGIT or a "-" character, ignore the cookie-av.
-    if (attribute_value.is_empty() || (!isdigit(attribute_value[0]) && (attribute_value[0] != '-')))
+    // 2. If the first character of the attribute-value is neither a DIGIT, nor a "-" character followed by a DIGIT,
+    //    ignore the cookie-av.
+    if (!is_ascii_digit(attribute_value[0]) && attribute_value[0] != '-')
```

Comment:
If `attribute_value[0]` is a `-`, it looks like the spec is saying `attribute_value[1]` should be a DIGIT.

Reply:
Right, but that is already covered by the steps below. If a cookie starts with an `-` followed by a non-digit, `attribute_value.to_number` will fail.